jnigen/lib/src/bindings/dart_bindings.dart

```diff
@@ -121,13 +128,34 @@ class CBasedDartBindingsGenerator extends BindingsGenerator {
 
     final resultGetter = getJValueAccessor(m.returnType);
     var wrapperExpr = '$sym(${actualArgs(m)}).$resultGetter';
-    wrapperExpr = toDartResult(wrapperExpr, m.returnType, returnTypeClass);
     final typeParamsWithExtend =
         dartTypeParams(m.typeParams, includeExtends: true);
     final params = getFormalArgs(c, m, resolver);
     s.write(
-        '$indent$ifStaticMethod $returnType $name$typeParamsWithExtend($params) '
-        '=> $wrapperExpr;\n');
+      '$indent$ifStaticMethod $returnType $name$typeParamsWithExtend($params) ',
+    );
+    if (m.asyncReturnType == null) {
+      wrapperExpr = toDartResult(wrapperExpr, m.returnType, returnTypeClass);
+      s.write('=> $wrapperExpr;\n');
+    } else {
+      s.write(
+        'async {\n'
```

Comment:
You an use multiline strings:

```dart
s.write('''
foo
  bar
  baz
bla
''');
```

Reply:
We never use multiline strings, so I'm being consistent here.

Reply:
We should change that, we can get rid of `\n` `$indent` etc!

(I don't mind being inconsistent for a while we go from one to the other.)
(Hm, doesn't seem like we have a lint for this.)